### DISCUSSION TITLE
One esp 01 connects and works with thonny but two others don't

### DISCUSSION THREAD

**Discussion by Strimis10** [micropython/micropython#9903] [Using MicroPython]
First of all sorry if this is in the wrong category.

so basically the title, I've got three esp01 one I've flashed micropython on and had it sitting in a project for three months. I also have two others that have just been sitting unused.
the one with micropython on it connects to thonny and works but the other two... thonny just shows the "Device is busy or does not respond" error, I can flash them using esp tool and the thonny version without getting any errors... 

I don't have a clue of what's wrong so I'd really appreciate the help :)

EDIT:

So when I install micropython this is what I get from esptool: 
"D:\downloads>python -m esptool -pCOM11 -b115200 write_flash --flash_mode dio --verify --flash_size detect 0 esp8266-1m-20220618-v1.19.1.bin
esptool.py v4.3
Serial port COM11
Connecting...
Detecting chip type... Unsupported detection protocol, switching and trying again...
Connecting...
Detecting chip type... ESP8266
Chip is ESP8266EX
Features: WiFi
Crystal is 26MHz
MAC: 40:91:51:4c:b7:e2
Stub is already running. No upload is necessary.
Configuring flash size...
Auto-detected Flash size: 1MB
Flash will be erased from 0x00000000 to 0x00090fff...
Flash params set to 0x0220
Compressed 591476 bytes to 392427...
Wrote 591476 bytes (392427 compressed) at 0x00000000 in 34.7 seconds (effective 136.5 kbit/s)...
Hash of data verified.

Leaving...
Verifying just-written flash...
(This option is deprecated, flash contents are now always read back after flashing.)
Flash params set to 0x0220
Verifying 0x90674 (591476) bytes @ 0x00000000 in flash against esp8266-1m-20220618-v1.19.1.bin...
-- verify OK (digest matched)
Hard resetting via RTS pin..."

but when I restart the esp and plug it back in:

"
D:\downloads>python -m esptool --port COM11 --baud 115200 verify_flash --diff no 0x40000 esp8266-1m-20220618-v1.19.1.bin
esptool.py v4.3
Serial port COM11
Connecting...
Detecting chip type... Unsupported detection protocol, switching and trying again...
Connecting...
Detecting chip type... ESP8266
Chip is ESP8266EX
Features: WiFi
Crystal is 26MHz
MAC: 40:91:51:4c:b7:e2
Stub is already running. No upload is necessary.
Configuring flash size...
Verifying 0x90674 (591476) bytes @ 0x00040000 in flash against esp8266-1m-20220618-v1.19.1.bin...
-- verify FAILED (digest mismatch)

A fatal error occurred: Verify failed."

**Comment by rkompass** [comment]
Did you try to select another port manually (in Execute / Select interpreter)?

**Reply by Strimis10** [reply]
yes, if not I'd have got "
Unable to connect to COM12: port not found

Process ended with exit code 1."

**Comment by karfas** [comment]
A quick search for "thonny" turns up, amongst others,  a message from @peterhinch:
https://github.com/micropython/micropython/discussions/9852#discussioncomment-4074418
> If I can offer another general comment. This isn't intended as a criticism of Thonny or any other IDE, but an IDE is another layer of software which has the potential for problems. I think is the point being made by @karfas.
> When confronted with a puzzling problem I recommend using the simplest, most reliable, tools available to access the device under test. In my opinion these are official mpremote and unofficial rshell; in extremis I use a terminal emulator - my choice is miniterm.py.

Someone else solved a similar problem: see https://github.com/micropython/micropython/discussions/9771#discussioncomment-3995641

My advice: use putty or the like to check if you can connect to the board (maybe try to use another USB cable).
When you get this stable, thonny will most likely also work.

**Reply by Strimis10** [reply]
I can connect to the board... and both thonny and putty can receive serial signals from the board if i hit the reset pin to ground but micropython won't work for some reason. and as I said I can easily upload micropython (1MB) to the board without any errors but when I plug it back in it's like it has nothing on it.

**Comment by karfas** [comment]
Could also be a reason: https://github.com/micropython/micropython/discussions/9523#discussioncomment-4123915

**Reply by scruss** [reply]
ESP01s don't have any USB interface chip, though

**Comment by robert-hh** [comment]
What are the flash sizes and types of the two non-func boards? Is that different to the functional one?

**Reply by Strimis10** [reply]
1MB all of them
